pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -124,15 +125,33 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 	if err != nil {
 		return nil, fmt.Errorf("unable to resolve rules for new global role: %w", err)
 	}
+
 	err = auth.ConfirmNoEscalation(request, clusterRules, "", a.grbResolver)
 	if err != nil {
+		hasEscalate, escErr := auth.RequestUserHasVerb(request, gvr, a.sar, escalateVerb, newGR.Name, "")
+		if escErr != nil {
+			logrus.Warnf("Failed to check for the 'escalate' verb on GlobalRoles: %v", escErr)
+			return admission.ResponseFailedEscalation(err.Error()), nil
+		}
+		if hasEscalate {
+			// if we have the escalate verb, no need to check global permissions
+			return admission.ResponseAllowed(), nil
+		}
 		return admission.ResponseFailedEscalation(err.Error()), nil
 	}
 
 	rules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(newGR)
 
 	err = auth.ConfirmNoEscalation(request, rules, "", a.resolver)
 	if err != nil {
+		hasEscalate, escErr := auth.RequestUserHasVerb(request, gvr, a.sar, escalateVerb, newGR.Name, "")
```

Comment:
The logic in `145`-`156` is largely the same as `129`-`141`. To ensure they are always in sync, I would extract them as a new `func` that is called at both places with diff params.

Reply:
Would you mind elaborating on your idea here? Given that most of this is error handling, I think that whatever I would do here would end up with similar duplicated code in the admit function (if err != nil{do stuff} is a min of 3 lines).

Reply:
The idea had two motivations:
- decrease the cognitive load by making it clear that the logic is the same, only the rules change.
- ensure that changes to the logic always happen at both places.

This was what I was thinking (instead of `129`-`158`):

```go
if res, err, ok := determineAdmission(request, clusterRules, "", a.grbResolver); ok {
  return res, err
}

rules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(newGR)
res, err, _ := determineAdmission(request, rules, "", a.grbResolver)
return res, err
```

Reply:
Ok, thanks for elaborating. Let me see what I can do.